### PR TITLE
Fix for the SequenceId ordering problem

### PR DESCRIPTION
**Pull request by ben-Draeger**
Currently the Database orders SequenceIds lexicographically, which
- is questionable as the SDC standards do not define any ordering upon SequenceIds, and
- may cause the behaviour of certain invariant test cases to change depending on which SequenceIds are used.

This Fix ensures that the SequenceIds retrieved from the Database are ordered by the timestamp of the first message that they were used in, making the order (and hence the behaviour of the invariant test cases) independent of the used SequenceIds.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Adherence to coding conventions
  * [x] Pull Request Assignee
  * [ ] Reviewer (maximilianpilz)
  * [ ] Reviewer (midttuna)
* Adherence to javadoc conventions
  * [x] Pull Request Assignee
  * [ ] Reviewer (maximilianpilz)
  * [ ] Reviewer (midttuna)
* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [ ] Reviewer (maximilianpilz)
  * [ ] Reviewer (midttuna)
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [ ] Reviewer (maximilianpilz)
  * [ ] Reviewer (midttuna)
* config update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [ ] Reviewer (maximilianpilz)
  * [ ] Reviewer (midttuna)
* SDCcc executable ran against a test device (if necessary)
  * [x] Pull Request Assignee
  * [ ] Reviewer (maximilianpilz)
  * [ ] Reviewer (midttuna)

### CODE REVIEW

**Review by midttuna — APPROVED** [review]

File: sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java [hunk 1]
```diff
@@ -1090,17 +1091,20 @@ public Stream<String> getUniqueSequenceIds() throws IOException {
...
-            criteria = criteriaBuilder.createQuery(String.class);
-            final Root<MdibVersionGroupEntity> mdibVersionGroupEntityRoot = criteria.from(MdibVersionGroupEntity.class);
-            criteria.select(mdibVersionGroupEntityRoot.get(MdibVersionGroupEntity_.sequenceId));
-            criteria.distinct(true);
+            messageContentQuery = criteriaBuilder.createQuery(String.class);
+            final Root<MessageContent> messageContentRoot = messageContentQuery.from(MessageContent.class);
+            messageContentQuery.select(
+                    messageContentRoot.join(MessageContent_.mdibVersionGroups).get(MdibVersionGroupEntity_.sequenceId));
+
+            messageContentQuery.orderBy(criteriaBuilder.asc(messageContentRoot.get(MessageContent_.nanoTimestamp)));
```
Comment (maximilianpilz) [inline sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java:1104]: Please add a unittest for this ordering.

**Review by maximilianpilz — CHANGES_REQUESTED** [review]
This change is impacted by https://github.com/Draegerwerk/SDCcc/issues/218, which has to be solved to guarantee the ordering is preserved on all levels.

File: sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java (hunk 1 above)
Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java:1104]: Done.

**Comment by ben-Draeger** [comment]
> This change is impacted by #218, which has to be solved to guarantee that the ordering is preserved on all levels.

I ensures that the used stream has ORDERED characteristics.

**Comment by ben-Draeger** [comment]
Closing this PR

**Pull request closed without merging**